libs/ledger-live-common/src/exchange/swap/completeExchange.ts

```diff
@@ -73,6 +73,10 @@ const completeExchange = (
         const errorsKeys = Object.keys(errors);
         if (errorsKeys.length > 0) throw errors[errorsKeys[0]]; // throw the first error
 
+        if (providerConfig.type !== "CEX") {
+          throw new Error(`Unsupported provider type ${providerConfig.type}`);
```

Comment:
Any idea how I can enter this state and test it @cjordan-ledger @sarneijim ?

Reply:
@ggilchrist-ledger AFAIK I don't believe it's currently possible to get to this line in the app, as DEX providers won't call this function. This has been added to ensure the code is type safe.